src/tree_data.c

```diff
@@ -2779,6 +2779,82 @@ lyd_target(const struct ly_path *path, const struct lyd_node *tree)
     return (struct lyd_node_term *)target;
 }
 
+/**
+ * @brief Compare schemas of two Compiled YANG data nodes.
+ *
+ * @param schema1 of first node.
+ * @param schema2 of second node.
+ * @return 0 if the schemas are equivalent otherwise 1.
+ */
+static ly_bool
+lyd_compare_schemas(const struct lysc_node *schema1, const struct lysc_node *schema2)
+{
+    if (schema1 == schema2) {
+        return 0;
+    } else if (!schema1 || !schema2) {
+        return 1;
+    }
+
```

Comment:
Here I would expect this code, with some comment even to explain it.
```C
if (schema1->module->ctx == schema2->module->ctx) {
    return 1;
}
```
Keep the same-context efficiency.

Reply:
why is this necessary? I thought that when the pointers `schema1 == schema2` are equal, they are in the same contexts. Isn't that so?

Reply:
But what about the situation when they are in the same context but `schema1 != schema2`? Right now the whole comparison for different context is performed, which is completely useless. What you should generally do is strictly distinguish in every comparison whether the contexts are the same or not, it save you these kinds of problems.

Reply:
ok, I get it